aws/resource_aws_apigatewayv2_api_mapping_test.go

```diff
@@ -0,0 +1,286 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/acm"
+	"github.com/aws/aws-sdk-go/service/apigatewayv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
```

Comment:
Should this have a sweeper, or will the resources get cleaned up when other resources are deleted?

Reply:
It's the `aws_acm_certificate` that was created that needs sweeping - I'll add a sweeper as that resource doesn't have one currently.
If I try to sweep APIs with API mappings defined I get:

```console
$ SWEEP=us-west-2 SWEEP_DIR=./aws/ SWEEPARGS=-sweep-run=aws_apigatewayv2_api make sweep
WARNING: This will destroy infrastructure. Use only in development accounts.
go test ./aws/ -v -sweep=us-west-2 -sweep-run=aws_apigatewayv2_api -timeout 60m
2020/04/18 17:56:38 [DEBUG] Running Sweepers for region (us-west-2):
2020/04/18 17:56:38 [DEBUG] Running Sweeper (aws_apigatewayv2_api) in region (us-west-2)
2020/04/18 17:56:38 [INFO] Building AWS auth structure
2020/04/18 17:56:38 [INFO] Setting AWS metadata API timeout to 100ms
2020/04/18 17:56:39 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2020/04/18 17:56:39 [INFO] AWS Auth provider used: "EnvProvider"
2020/04/18 17:56:39 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2020/04/18 17:56:40 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2020/04/18 17:56:41 [INFO] Deleting API Gateway v2 API: bvowp2c4b5
2020/04/18 17:56:41 [ERROR] error deleting API Gateway v2 API (bvowp2c4b5): BadRequestException: Deleting RestApi bvowp2c4b5 failed. Please remove all base path mappings related to the RestApi in your domains.
2020/04/18 17:56:41 [ERROR] Error running Sweeper (aws_apigatewayv2_api) in region (us-west-2): 1 error occurred:
	* error deleting API Gateway v2 API (bvowp2c4b5): BadRequestException: Deleting RestApi bvowp2c4b5 failed. Please remove all base path mappings related to the RestApi in your domains.

FAIL	github.com/terraform-providers/terraform-provider-aws/aws	3.718s
FAIL
GNUmakefile:19: recipe for target 'sweep' failed
make: *** [sweep] Error 1
```

That error message seems to be a holdover from API Gateway v1 but it means that domain names need to be deleted first, so I've added a dependency to the `aws_apigatewayv2_api` sweeper:

```console
$ SWEEP=us-west-2 SWEEP_DIR=./aws/ SWEEPARGS=-sweep-run=aws_apigatewayv2_api make sweep
WARNING: This will destroy infrastructure. Use only in development accounts.
go test ./aws/ -v -sweep=us-west-2 -sweep-run=aws_apigatewayv2_api -timeout 60m
2020/04/18 18:14:38 [DEBUG] Running Sweepers for region (us-west-2):
2020/04/18 18:14:38 [DEBUG] Sweeper (aws_apigatewayv2_api) has dependency (aws_apigatewayv2_domain_name), running..
2020/04/18 18:14:38 [DEBUG] Running Sweeper (aws_apigatewayv2_domain_name) in region (us-west-2)
2020/04/18 18:14:38 [INFO] Building AWS auth structure
2020/04/18 18:14:38 [INFO] Setting AWS metadata API timeout to 100ms
2020/04/18 18:14:40 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2020/04/18 18:14:40 [INFO] AWS Auth provider used: "EnvProvider"
2020/04/18 18:14:40 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2020/04/18 18:14:40 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2020/04/18 18:14:41 [INFO] Deleting API Gateway v2 domain name: example-ws-api.example.com
2020/04/18 18:14:42 [DEBUG] Running Sweeper (aws_apigatewayv2_api) in region (us-west-2)
2020/04/18 18:14:42 [INFO] Building AWS auth structure
2020/04/18 18:14:42 [INFO] Setting AWS metadata API timeout to 100ms
2020/04/18 18:14:43 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2020/04/18 18:14:43 [INFO] AWS Auth provider used: "EnvProvider"
2020/04/18 18:14:43 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2020/04/18 18:14:43 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2020/04/18 18:14:44 [INFO] Deleting API Gateway v2 API: m36pbck0ii
2020/04/18 18:14:46 [DEBUG] Sweeper (aws_apigatewayv2_domain_name) already ran in region (us-west-2)
2020/04/18 18:14:46 Sweeper Tests ran successfully:
	- aws_apigatewayv2_domain_name
	- aws_apigatewayv2_api
ok  	github.com/terraform-providers/terraform-provider-aws/aws	7.842s
```